### PR TITLE
Mention Spatial -> Node3D rename in Introduction docs

### PR DESCRIPTION
**Pull request by paddy-exe**
Fixes https://github.com/godotengine/godot-docs/issues/7337

I added the renaming in "introduction to 3D" and "key_concepts_overview"

### CODE REVIEW

File: getting_started/introduction/key_concepts_overview.rst [hunk 1]
```diff
@@ -46,7 +46,9 @@ It is made of a ``CharacterBody2D`` node named "Character", a ``Sprite2D``, a
 ``Camera2D``, and a ``CollisionShape2D``.
 
 .. note:: The node names end with "2D" because this is a 2D scene. Their 3D
-          counterparts have names that end with "3D".
+          counterparts have names that end with "3D". Be aware that there
+          was a renaming of "Spatial" Nodes to "Node3D" starting with Godot
+          4.
```
Comment (clayjohn) [inline getting_started/introduction/key_concepts_overview.rst:51]:
```suggestion
          counterparts have names that end with "3D". Be aware that "Spatial"
          Nodes are now called "Node3D" starting with Godot 4.
```
I think this just reads a little cleaner as it avoids passive voice

File: tutorials/3d/introduction_to_3d.rst [hunk 2]
```diff
@@ -24,6 +24,10 @@ Node3D node
 Following this reasoning, the 3D engine uses the :ref:`Node3D <class_Node3D>`
 node for everything 3D.
 
+.. note:: Be aware that there was a renaming of "Spatial" Nodes to
+   "Node3D" starting Godot 4. Any Godot 3.x references about "Spatial"
+   Nodes refer to "Node3D" in Godot 4.
```
Comment (clayjohn) [inline tutorials/3d/introduction_to_3d.rst:29]:
```suggestion
.. note:: Be aware that "Spatial" Nodes are now called "Node3D"
   starting with Godot 4.. Any Godot 3.x references to "Spatial"
   Nodes refer to "Node3D" in Godot 4.
```

**Review by clayjohn — COMMENTED** [review]
Looks good! Just a small grammar tweak that I think helps the readability

File: tutorials/3d/introduction_to_3d.rst (hunk 2 above)
Comment (paddy-exe) [reply tutorials/3d/introduction_to_3d.rst]: Is the double dot after `with Godot 4` intentional?

**Comment by paddy-exe** [comment]
Updated to your suggestions

**Comment by paddy-exe** [comment]
I added a section to the class reference of the Node3D as suggested in the issue. I think this makes sense since some people won't  be looking in the getting started sections necesserily.

File: tutorials/3d/introduction_to_3d.rst (hunk 2 above)
Comment (mhilbrunner) [reply tutorials/3d/introduction_to_3d.rst]: I doubt it :)

File: classes/class_node3d.rst [hunk 3]
```diff
@@ -27,6 +27,8 @@ Affine operations (rotate, scale, translate) happen in parent's local coordinate
 
 \ **Note:** Unless otherwise specified, all methods that have angle parameters must have angles specified as *radians*. To convert degrees to radians, use :ref:`@GlobalScope.deg_to_rad<class_@GlobalScope_method_deg_to_rad>`.
 
+\ **Note:** Be aware that "Spatial" Nodes are now called "Node3D starting with Godot 4. Any Godot 3.x references to "Spatial" Nodes refer to "Node3D" in Godot 4.
```
[the remark this answers is not in the document]
Comment (paddy-exe) [reply classes/class_node3d.rst]: Removed the changes from this file and will create a new PR in the main repo then

**Review by clayjohn — APPROVED** [review]
Looks great to me! Thank you!

**Review by Piralein — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**